miniml/compiler/compile.scm

```diff
@@ -529,6 +550,30 @@
           (else #nil)
           )))
 
+(define (symbol-char? c)
+  (member c (list #\! #\$ #\% #\& #\* #\+ #\- #\. #\/ #\: #\< #\= #\> #\? #\@ #\^ #\| #\~) char=?))
```

Comment:
Sorry for the nitpick, but my screen is not as wide as yours apparently, so I would be in favor of not going above 80-100 chars per line. I would break a line somewhere here.

Reply:
Note: this code could also be more readable if you used `string-index` to search within a string literal, rather than a long and not-so-easy-to-read list of literal characters.

Reply:
Indeed, I hadn't thought about using a string! It is much more readable now.